packages/components/data-table-manager/src/data-table-manager.tsx

```diff
@@ -0,0 +1,192 @@
+   * the default `itemRenderer` prop of the table.
+   */
+  renderItem?: (row: ReactNode, isRowCollapsed: boolean) => ReactNode;
+
+  /**
+   * Use this prop to place an `Icon` or `IconButton` on the left of the column label.
+   * It is advised to place these types of components through this prop instead of `label`,
+   * in order to properly position and align the elements.
+   * This is particularly useful for medium-sized icons which require more vertical space than the typography.
+   */
+  headerIcon?: ReactNode;
+
+  /**
+   * Set this to `true` to allow text content of this cell to be truncated with an ellipsis,
+   * instead of breaking into multiple lines.
+   * <br>
+   * NOTE: when using this option, it is recommended to specify a `width` for the column, because
+   * if the table doesn't have enough space for all columns, it will start clipping the columns
+   * with _truncated_ content, and if no `width` is set (or the value is set `auto` -- the default)
+   * it can shrink until the column disappears completely.
+   * By enforcing a minimum width for these columns, the table will respect them and grow horizontally,
+   * adding scrollbars if needed.
+   *
+   * @@defaultValue@@: false
+   */
+  isTruncated?: boolean;
+
+  /**
+   * Set this to `true` to show a sorting button, which calls `onSortChange` upon being clicked.
+   * You should enable this flag for every column you want to be able to sort.
+   * When at least one column is sortable, the table props `sortBy`, `sortDirection` and `onSortChange` should be provided.
+   *
+   * @@defaultValue@@: false
+   */
+  isSortable?: boolean;
+
+  /**
+   * Set this to `true` to prevent this column from being manually resized by dragging
+   * the edge of the header with a mouse.
+   *
+   * @@defaultValue@@: false
+   */
+  disableResizing?: boolean;
+
+  /**
+   * Set this to `true` to prevent click event propagation for this cell.
+   * You might want this if you need the column to have its own call-to-action or input while
+   * the row also has a defined `onRowClick`.
+   *
+   * @@defaultValue@@: false
+   */
+  shouldIgnoreRowClick?: boolean;
+};
+
+type TDataTableManagerProps = {
+  /**
+   * Each object requires a unique `key` which should correspond to property key of
+   * the items of `rows` that you want to render under this column, and a `label`
+   * which defines the name shown on the header.
+   * The list of columns to be rendered.
+   * Each column can be customized (see properties below).
+   */
+  columns: TColumProps[];
+
+  /**
+   * Any React node. Usually you want to render the `<DataTable>` component.
+   * <br>
+   * Note that the child component will implicitly receive the props `columns` and `isCondensed` from the `<DataTableManager>`.
+   */
+  children: JSX.Element;
+
+  /**
+   * The managed display settings of the table.
+   */
+  displaySettings?: TDataTableSettingsProps['displaySettings'];
```

Comment:
Not sure the best way to approach this.
Both `displaySettings` and `columnManager` props were just reflected from the private `DataTableSettings` component so here I'm just using the definitions in it but this will not generate a meaningful README.
Maybe I should type that objects here as well so we get a right README or should we go for extending the readme generation script to handle this situation (not sure about its complexity because it would involve parsing a different file that the one parsed when generating the documentation).

Reply:
I'm not sure as well, another option is to add the link to the docs for the TDataTableSettingsProps just like we did for react-select.

Reply:
Yes, but the problem is that `DataTableSettingsProps` is a private component so no README for it exists in the documentation where to link to.

Reply:
We will need to update the READMEs generator to take this cases into account.
Until them, I disabled the auto-generation of this component README by renaming the `docs` folder (the generator will ignore it).